src/bridge/mod.rs

```diff
@@ -44,6 +47,84 @@ fn neovim_instance() -> Result<NeovimInstance> {
     }
 }
 
+/// Takes the --cmd or -c argument and returns the command to be executed.
+#[cfg(target_os = "macos")]
+fn handle_command_arg(position: usize, args: Vec<String>) -> String {
+    args.get(position + 1).cloned().unwrap_or_default()
+}
+
+/// Takes the valid path argument and returns the startup directory.
+#[cfg(target_os = "macos")]
+fn handle_command_arg_as_path_or_default(args: &mut Vec<String>) -> String {
+    args.retain(|arg| is_valid_path(arg));
+
+    let path = args.first().cloned().unwrap_or_default();
+    let startup_directory = get_startup_directory(&path);
+
+    format!(
+        "if g:neovide_ntty == v:true | chdir {} | endif",
+        startup_directory
+    )
+}
+
+/// The function `setup_tty_startup_directory` sets up the startup directory for
+/// a new TTY Neovim session on macOS platform, such as Finder, Dock, or even
+/// external programs like Neohub.
+///
+/// Any nvim --cmd or -c argument is handled as a command to be executed.
+///
+/// Conditions:
+///
+/// - Is not a TTY session.
+/// - Argument is a directory, it becomes the startup directory.
+/// - Argument is a file, its parent directory becomes the startup directory.
+/// - Neither directory nor file, $HOME is used.
+#[cfg(target_os = "macos")]
+pub async fn setup_tty_startup_directory(
+    nvim: &Neovim<NeovimWriter>,
+) -> Result<(), Box<CallError>> {
+    use self::command::is_tty;
+
+    if is_tty() {
+        return Ok(());
+    }
+
+    let neovim_args = SETTINGS.get::<CmdLineSettings>().neovim_args;
+
+    let cmd_arg = neovim_args
+        .iter()
+        .rposition(|arg| arg == "--cmd" || arg == "-c");
+
+    let mut cmd = handle_command_arg_as_path_or_default(&mut neovim_args.clone());
+
+    if let Some(pos) = cmd_arg {
+        cmd = format!("{} | {}", cmd, handle_command_arg(pos, neovim_args));
+    }
+
+    nvim.command(cmd.as_str()).await
+}
+
+#[cfg(target_os = "macos")]
+fn get_startup_directory(path: &str) -> String {
+    use std::path::{Path, PathBuf};
+    match path {
+        arg if PathBuf::from(&arg).is_dir() => arg.to_string(),
+        arg if PathBuf::from(&arg).is_file() => Path::new(&arg)
+            .parent()
+            .unwrap_or_else(|| Path::new(TTY_STARTUP_DIRECTORY))
+            .to_string_lossy()
+            .to_string(),
+        _ => TTY_STARTUP_DIRECTORY.to_string(),
+    }
+}
```

Comment:
I think the home dir should be the default, unless a file or directory is opened, on all platforms, not just macOS,  when running from the GUI.

But that looks like it's already the case by default, is it not on macOS? What is it if you don't change it here?

Reply:
on macOS if you launch from Finder or Dock the default is the root path `/`

Reply:
that's why I made it specifically to macOS